### PR TITLE
Add CloneContent Permission as Content Type dynamic permission

### PR DESCRIPTION
**Pull request by chinasqzl**
Adds the ability to set CloneContent permission on each Content Type.
Closes [#7441](https://github.com/OrchardCMS/OrchardCore/issues/7441)
image

### CODE REVIEW

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=7442) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=7442) before we can accept your contribution.<br/><br/>:x: chinasqzl [sign now](https://cla.dotnetfoundation.org/OrchardCMS/OrchardCore?pullRequest=7442)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/OrchardCMS/OrchardCore?pullRequest=7442) it.</sub>

File: src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TaxonomyContentsAdminListDisplayDriver.cs [hunk 1]
```diff
@@ -46,8 +51,18 @@ public override async Task<IDisplayResult> EditAsync(ContentOptionsViewModel mod
...
+                var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(model.SelectedContentType);
+                var fieldDefinitions = contentTypeDefinition
+                    .Parts.SelectMany(x => x.PartDefinition.Fields.Where(f => f.FieldDefinition.Name == nameof(TaxonomyField)));
+                var fieldTaxonomyContentItemIds = fieldDefinitions.Select(x => x.GetSettings<TaxonomyFieldSettings>().TaxonomyContentItemId);
+                taxonomyContentItemIds = taxonomyContentItemIds.Intersect(fieldTaxonomyContentItemIds).ToArray();
+            }
+
             var results = new List<IDisplayResult>();
-            var taxonomies = await _contentManager.GetAsync(settings.TaxonomyContentItemIds);
+            var taxonomies = await _contentManager.GetAsync(taxonomyContentItemIds);
```
Comment (Piedone) [inline src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TaxonomyContentsAdminListDisplayDriver.cs:65]: These changes seem unrelated to the issue. The problem is that you're merging from your fork's dev branch; it's better to create a branch for every patch you're working on. As a stopgap solution, you can just revert the commit where you've made this change.

**Review by Piedone — CHANGES_REQUESTED** [review]

File: src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TaxonomyContentsAdminListDisplayDriver.cs (hunk 1 above)
Comment (chinasqzl) [reply src/OrchardCore.Modules/OrchardCore.Taxonomies/Drivers/TaxonomyContentsAdminListDisplayDriver.cs]: Restored

**Review by Piedone — APPROVED** [review]
